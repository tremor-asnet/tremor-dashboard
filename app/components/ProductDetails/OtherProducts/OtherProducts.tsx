"use client";

// Components
import {
  CustomAvatarName,
  CustomNumberFormat,
} from "@/components/Table/common";
import { DataGrid } from "@/components";

// Types
import { Product, ColumnType } from "@/types";

// Constants
import { ProgressBar } from "@tremor/react";

//Styles
import "@/styles/products.css";

interface OtherProductsProps {
  products: Product[];
  isAvailable: string;
  keyword: string;
  className?: string;
  hasPagination?: boolean;
  hasSort?: boolean;
}

const OtherProducts = ({
  products,
  isAvailable,
  keyword,
  className = "",
  hasPagination = true,
  hasSort = true,
}: OtherProductsProps) => {
  // Other Product Table Props
  const columns: ColumnType<Product>[] = [
    {
      key: "product",
      title: "Product",
      customNode: (_, { productName, image }) => (
        <CustomAvatarName avatar={image} text={productName} />
      ),
    },
    {
      key: "price",
      title: "Price",
      customNode: (_, { price }) => <CustomNumberFormat value={price} />,
    },
    {
      key: "isAvailable",
      title: "Availability",
      customNode: (_, { isAvailable }) => (
        <p className="text-xs dark:text-lighter font-semibold product-availability">
          {isAvailable ? (
            <ProgressBar value={80} color="green" className="w-32" />
          ) : (
            <ProgressBar value={60} color="orange" className="w-32" />
          )}
        </p>
      ),
    },
    {
      key: "id",
      title: "Id",
    },
  ];

  return (
    <DataGrid
      data={products}
      columns={columns}
      filterBy={isAvailable}
      keyword={keyword}
      className={className}
      hasPagination={hasPagination}
      hasSort={hasSort}
    />
  );
};

export default OtherProducts;
