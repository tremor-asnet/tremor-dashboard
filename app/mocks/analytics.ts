export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
