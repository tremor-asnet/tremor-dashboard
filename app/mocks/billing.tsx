// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
