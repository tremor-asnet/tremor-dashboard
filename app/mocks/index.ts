export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
