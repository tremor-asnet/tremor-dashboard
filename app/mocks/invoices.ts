import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
