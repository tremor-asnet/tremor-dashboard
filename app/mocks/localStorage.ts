// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
