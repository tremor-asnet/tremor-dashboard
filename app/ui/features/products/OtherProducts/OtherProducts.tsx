"use client";

// Components
import {
  DataGrid,
  CustomAvatarName,
  CustomNumberFormat,
} from "@/ui/components";
import { Text } from "@tremor/react";
import Link from "next/link";

// Types
import { Product, ColumnType } from "@/types";

// Constants
import { ROUTES } from "@/constants";

// Styles
import "@/styles/products.css";

export const OtherProducts = ({ products }: { products: Product[] }) => {
  // Other Product Table Props
  const columns: ColumnType<Product>[] = [
    {
      key: "id",
      title: "Id",
      customNode: (_, { id }) => (
        <Link
          prefetch={true}
          href={`${ROUTES.PRODUCT_LIST}/${id}`}
          className="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline">
          &#35;{id}
        </Link>
      ),
    },
    {
      key: "product",
      title: "Product",
      customNode: (_, { productName, image }) => (
        <CustomAvatarName avatar={image} text={productName} />
      ),
    },
    {
      key: "price",
      title: "Price",
      customNode: (_, { price }) => <CustomNumberFormat value={price} />,
    },
    {
      key: "isAvailable",
      title: "Is Available",
      customNode: (_, { quantity }) => (
        <Text className="text-xs dark:text-lighter font-semibold">
          {quantity && quantity > 0 ? "Yes" : "No"}
        </Text>
      ),
    },
  ];

  return (
    <DataGrid
      data={products}
      columns={columns}
      className="!shadow-none"
      hasPagination={false}
    />
  );
};

export default OtherProducts;
