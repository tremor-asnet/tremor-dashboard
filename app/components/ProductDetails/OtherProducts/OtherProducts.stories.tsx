import type { Meta, StoryObj } from "@storybook/react";

// Components
import OtherProducts from "./OtherProducts";

// Mocks
import { MOCK_PRODUCTS } from "@/mocks";

const meta = {
  title: "Components/OtherProducts",
  component: OtherProducts,
  tags: ["autodocs"],
} as Meta<typeof OtherProducts>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => (
    <OtherProducts
      products={MOCK_PRODUCTS}
      isAvailable="isAvailable"
      keyword="productName"
    />
  ),
};
