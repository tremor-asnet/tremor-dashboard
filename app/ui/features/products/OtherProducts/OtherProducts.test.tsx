import { render } from "@testing-library/react";

// Mocks
import { MOCK_PRODUCTS } from "@/mocks";

// Components
import OtherProducts from "./OtherProducts";

jest.mock("next/navigation", () => {
  const actual = jest.mock("next/navigation");
  return {
    ...actual,
    useRouter: jest.fn(() => ({
      push: jest.fn(),
      replace: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({})),
    usePathname: jest.fn(),
  };
});

describe("Other Products Table Testing", () => {
  it("should match snapshot", () => {
    const { container } = render(<OtherProducts products={MOCK_PRODUCTS} />);

    expect(container).toMatchSnapshot();
  });
});
