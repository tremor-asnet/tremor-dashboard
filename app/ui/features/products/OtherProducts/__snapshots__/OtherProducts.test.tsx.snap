// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Products Table Testing should match snapshot 1`] = `
<div>
  <div
    class="tremor-Card-root relative w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-0 border-none ring-0 dark:bg-dark-tremor-primary overflow-x-auto !shadow-none"
  >
    <div
      class="flex flex-col items-start justify-start my-2"
    >
      <div
        class="tremor-Table-root overflow-auto w-full"
      >
        <table
          class="tremor-Table-table w-full text-tremor-default text-tremor-content dark:text-dark-tremor-content"
        >
          <thead
            class="tremor-TableHead-root text-left text-tremor-content dark:text-dark-tremor-content"
          >
            <tr
              class="tremor-TableRow-row"
            >
              <th
                class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold opacity-70 uppercase cursor-pointer print:!text-primary dark:print:!text-primary dark:print:opacity-100"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                >
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                  >
                    Id
                  </div>
                </div>
              </th>
              <th
                class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold opacity-70 uppercase cursor-pointer print:!text-primary dark:print:!text-primary dark:print:opacity-100"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                >
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                  >
                    Product
                  </div>
                </div>
              </th>
              <th
                class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold opacity-70 uppercase cursor-pointer print:!text-primary dark:print:!text-primary dark:print:opacity-100"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                >
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                  >
                    Price
                  </div>
                </div>
              </th>
              <th
                class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold opacity-70 uppercase cursor-pointer print:!text-primary dark:print:!text-primary dark:print:opacity-100"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                >
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                  >
                    Is Available
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="tremor-TableBody-root align-top divide-y divide-tremor-border dark:divide-dark-tremor-border"
          >
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <a
                  class="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline"
                  href="/product-list/230019"
                >
                  #
                  230019
                </a>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <div
                  class="flex justify-start items-center w-48"
                >
                  <img
                    alt="Christopher Knight Home"
                    class="w-6 h-6 min-w-[24px] rounded-full mr-2"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                  <p
                    class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer truncate"
                  >
                    Christopher Knight Home
                  </p>
                </div>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-revenue"
                >
                  $89.53
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold"
                >
                  No
                </p>
              </td>
            </tr>
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <a
                  class="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline"
                  href="/product-list/87120"
                >
                  #
                  87120
                </a>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <div
                  class="flex justify-start items-center w-48"
                >
                  <img
                    alt="Bar Height Swivel Barstool"
                    class="w-6 h-6 min-w-[24px] rounded-full mr-2"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-pink.686a7ae1.jpeg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-pink.686a7ae1.jpeg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-pink.686a7ae1.jpeg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                  <p
                    class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer truncate"
                  >
                    Bar Height Swivel Barstool
                  </p>
                </div>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-revenue"
                >
                  $99.99
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold"
                >
                  No
                </p>
              </td>
            </tr>
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <a
                  class="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline"
                  href="/product-list/2456772"
                >
                  #
                  2456772
                </a>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <div
                  class="flex justify-start items-center w-48"
                >
                  <img
                    alt="Christopher Knight Home"
                    class="w-6 h-6 min-w-[24px] rounded-full mr-2"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                  <p
                    class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer truncate"
                  >
                    Christopher Knight Home
                  </p>
                </div>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-revenue"
                >
                  $89.53
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold"
                >
                  No
                </p>
              </td>
            </tr>
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <a
                  class="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline"
                  href="/product-list/1992"
                >
                  #
                  1992
                </a>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <div
                  class="flex justify-start items-center w-48"
                >
                  <img
                    alt="Modern Square"
                    class="w-6 h-6 min-w-[24px] rounded-full mr-2"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-wood.d127f22b.jpeg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-wood.d127f22b.jpeg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-wood.d127f22b.jpeg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                  <p
                    class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer truncate"
                  >
                    Modern Square
                  </p>
                </div>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-revenue"
                >
                  $59.99
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold"
                >
                  No
                </p>
              </td>
            </tr>
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <a
                  class="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline"
                  href="/product-list/412301"
                >
                  #
                  412301
                </a>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <div
                  class="flex justify-start items-center w-48"
                >
                  <img
                    alt="Signature Design by Ashley"
                    class="w-6 h-6 min-w-[24px] rounded-full mr-2"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-steel.dd1d3892.jpeg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-steel.dd1d3892.jpeg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fchair-steel.dd1d3892.jpeg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                  <p
                    class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer truncate"
                  >
                    Signature Design by Ashley
                  </p>
                </div>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-revenue"
                >
                  $129.00
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
              >
                <p
                  class="text-tremor-default text-tremor-content text-xs dark:text-lighter font-semibold"
                >
                  No
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
